image.go

```diff
@@ -0,0 +1,364 @@
+	// VirtType             VirtType `mapstructure:"virt_type"`
+	Arch                 string      `mapstructure:"arch"`
+	Autoinstall          string      `mapstructure:"autoinstall"`
+	Breed                string      `mapstructure:"breed"`
+	File                 string      `mapstructure:"file"`
+	ImageType            string      `mapstructure:"image_type"`
+	NetworkCount         int         `mapstructure:"network_count"`
+	OsVersion            string      `mapstructure:"os_version"`
+	BootLoaders          []string    `mapstructure:"boot_loaders"`
+	Menu                 string      `mapstructure:"menu"`
+	VirtAutoBoot         bool        `mapstructure:"virt_auto_boot"`
+	VirtBridge           string      `mapstructure:"virt_bridge"`
+	VirtCpus             int         `mapstructure:"virt_cpus"`
+	VirtDiskDriver       string      `mapstructure:"virt_disk_driver"`
+	VirtFileSize         interface{} `mapstructure:"virt_file_size"`
+	VirtPath             string      `mapstructure:"virt_path"`
+	VirtRam              int         `mapstructure:"virt_ram"`
+	VirtType             string      `mapstructure:"virt_type"`
+	SupportedBootLoaders []string    `mapstructure:"supported_boot_loaders"`
+
+	Client
+}
+
+func convertRawImage(name string, xmlrpcResult interface{}) (*Image, error) {
+	var image Image
+
+	if xmlrpcResult == "~" {
+		return nil, fmt.Errorf("profile %s not found", name)
+	}
+
+	decodeResult, err := decodeCobblerItem(xmlrpcResult, &image)
+	if err != nil {
+		return nil, err
+	}
+
+	return decodeResult.(*Image), nil
+}
+
+func convertRawImagesList(xmlrpcResult interface{}) ([]*Image, error) {
+	var images []*Image
+
+	for _, d := range xmlrpcResult.([]interface{}) {
+		distro, err := convertRawImage("unknown", d)
+		if err != nil {
+			return nil, err
+		}
+		images = append(images, distro)
+	}
+
+	return images, nil
+}
+
+// GetImages returns all images in Cobbler.
+func (c *Client) GetImages() ([]*Image, error) {
+	result, err := c.Call("get_images", "-1", c.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	return convertRawImagesList(result)
+}
+
+func (c *Client) ListImageNames() ([]string, error) {
+	return c.GetItemNames("image")
+}
+
+// GetImage returns a single image obtained by its name.
+func (c *Client) GetImage(name string) (*Image, error) {
+	result, err := c.Call("get_image", name, c.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	return convertRawImage(name, result)
+}
+
+// CreateImage creates a profile.
+func (c *Client) CreateImage(image Image) (*Image, error) {
+	// To create an image via the Cobbler API, first call new_image to obtain an ID
+	result, err := c.Call("new_image", c.Token)
+	if err != nil {
+		return nil, err
+	}
+	newID := result.(string)
+	// Set the value of all fields
+	item := reflect.ValueOf(&image).Elem()
+	if err := c.updateCobblerFields("profile", item, newID); err != nil {
+		return nil, err
+	}
+
+	// Save the final profile
+	if err = c.SaveImage(newID, "new"); err != nil {
```

Comment:
From the diff to the previously review version this call changed from save_profile to SaveImage. According to the comment as well this should be SaveProfile instead.

Reply:
Yes but actually both are wrong. This is the method to create an image. As such this is a copy & paste error from me.